scripts/dockerrun.sh

```diff
@@ -87,8 +87,43 @@ fi
 #  VSO
 [ ! -z "$BUILD_BUILDID" ] && DOTNET_BUILD_CONTAINER_NAME="${BUILD_BUILDID}-${BUILD_BUILDNUMBER}"
 
+# Executes a command and retries if it fails.
+# NOTE: This function is the exact copy from init-docker.sh.
+# Reason for not invoking init.docker.sh directly is since that script 
+# also performs cleanup, which we do not want in this case.
```

Comment:
@dagood, good point on the split.  That being said, wouldn't this also require a change so that the build tools are available at this point in the build for the dev-workflow?  This comes back to the problem that it is not drop dead easy to share common components.  Perhaps the approach of auto-coping the shared scripts into each repo would be the best here.